### PR TITLE
[Bugfix](primary-key) fix calc delete bitmap bug in concurrent memtable flush

### PR DESCRIPTION
**Pull request by yixiutt**
# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/olap/delta_writer.cpp [hunk 1]
```diff
@@ -281,7 +283,7 @@ Status DeltaWriter::wait_flush() {
 }
 
 void DeltaWriter::_reset_mem_table() {
-    if (_tablet->enable_unique_key_merge_on_write()) {
+    if (_delete_bitmap == nullptr) {
```
Comment (liaoxin01) [inline be/src/olap/delta_writer.cpp:286]:
```suggestion
    if (_tablet->enable_unique_key_merge_on_write() && _delete_bitmap == nullptr) {
```

Comment (yixiutt) [reply be/src/olap/delta_writer.cpp]: done

**Review by liaoxin01 — APPROVED** [review]
LGTM

**Review by liaoxin01 — DISMISSED** [review]
LGTM

File: be/src/olap/memtable_flush_executor.h [hunk 2]
```diff
@@ -107,7 +107,7 @@ class MemTableFlushExecutor {
     void init(const std::vector<DataDir*>& data_dirs);
 
     Status create_flush_token(std::unique_ptr<FlushToken>* flush_token, RowsetTypePB rowset_type,
-                              bool is_high_priority);
+                              bool merge_on_write, bool is_high_priority);
```
Comment (zhannngchen) [inline be/src/olap/memtable_flush_executor.h:110]: the new parameter should rename to `serial_mode`, MoW is a feature of unique key table, use `merge_on_write` here is confusing.

File: be/src/olap/memtable_flush_executor.cpp [hunk 3]
```diff
@@ -132,10 +132,12 @@ void MemTableFlushExecutor::init(const std::vector<DataDir*>& data_dirs) {
...
 
 // NOTE: we use SERIAL mode here to ensure all mem-tables from one tablet are flushed in order.
+// unique key merge on write should flush serial cause calc delete bitmap should load segment serial
 Status MemTableFlushExecutor::create_flush_token(std::unique_ptr<FlushToken>* flush_token,
-                                                 RowsetTypePB rowset_type, bool is_high_priority) {
+                                                 RowsetTypePB rowset_type, bool merge_on_write,
+                                                 bool is_high_priority) {
     if (!is_high_priority) {
-        if (rowset_type == BETA_ROWSET) {
+        if (rowset_type == BETA_ROWSET && !merge_on_write) {
```
Comment (zhannngchen) [inline be/src/olap/memtable_flush_executor.cpp:140]: L150 should also consider the parameter.

File: be/src/olap/memtable_flush_executor.cpp [hunk 4]
```diff
@@ -132,10 +132,12 @@ void MemTableFlushExecutor::init(const std::vector<DataDir*>& data_dirs) {
 }
 
 // NOTE: we use SERIAL mode here to ensure all mem-tables from one tablet are flushed in order.
+// unique key merge on write should flush serial cause calc delete bitmap should load segment serial
```
Comment (zhannngchen) [inline be/src/olap/memtable_flush_executor.cpp:135]: This comment should move to `be/src/olap/delta_writer.cpp:L143`

**Review by zhannngchen — COMMENTED** [review]
the `primary-key` in title should change to `unique-key-merge-on-write`

File: be/src/olap/memtable_flush_executor.cpp (hunk 3 above)
Comment (yixiutt) [reply be/src/olap/memtable_flush_executor.cpp]: done

File: be/src/olap/memtable_flush_executor.cpp (hunk 4 above)
Comment (yixiutt) [reply be/src/olap/memtable_flush_executor.cpp]: done

**Review by zhannngchen — APPROVED** [review]
LGTM

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**